ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1337 @@
+            self.log.error(msg)
+            raise
+
+        self.h5py = h5py
+        self.file = h5py.File(self.input_url)
+
+#         if MAGICEventSource._count > 0:
+#             self.log.warn("Only one MAGIC raw event_source allowed at a time. "
+#                           "Previous hdf5 file will be closed.")
+#             self.pyhessio.close_file()
+#         MAGICEventSource._count += 1
+
+
+    @staticmethod
+    def is_compatible(file_path):
+        import h5py
+        # check general format:
+        if not h5py.is_hdf5(file_path):
+            return False
+        # crude check if hdf5 file contains MAGIC raw data:
+        with h5py.File(file_path, "r") as file:
+            intrument_attr = file.attrs['instrument']
+            if intrument_attr.tostring() == b"MAGIC":
+                return True
+            else:
+                return False
+            
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.file.close()
+
+    def _generator(self):
+        with self.h5py.File(self.input_url, "r") as file:
+            # the container is initialized once, and data is replaced within
+            # it after each yield
+            counter = 0
+            #eventstream = file.move_to_next_event()
+            data = DataContainer()
+            data.meta['origin'] = "MAGIC"
+
+            # some hessio_event_source specific parameters
+            data.meta['input_url'] = self.input_url
+            data.meta['max_events'] = self.max_events
+
+            events_per_tel = [[],[]]
+
+            if not file.attrs['RunType'] == b"Data":
+                data.meta['is_simulation'] = True
+                dt = np.float
+            else:
+                data.meta['is_simulation'] = False
+                dt = np.int16
+                
+            # check which telescopes have data:
+            
+            if file.attrs['dl_export'] == b"dl1":
+                eventstream = file['dl1/event_id']
+                if file.attrs['data format'] == b"mono M1":
+                    tels_in_file = ["M1"]
+                elif file.attrs['data format'] == b"mono M2":
+                    tels_in_file = ["M2"]
+                elif file.attrs['data format'] == b"stereo":
+                    tels_in_file = ["M1", "M2"]
+                
+            elif file.attrs['dl_export'] == b"r0":
+                if list(file.keys()) == ["MAGIC1_rawdata"]:
+                    eventstream = file['MAGIC1_rawdata/EvtHeader/StereoEvtNumber']
+                    events_per_tel[0] = np.array(eventstream, dtype = dt)
+                    tels_in_file = [1]
+                elif list(file.keys()) == ["MAGIC2_rawdata"]:
+                    eventstream = file['MAGIC2_rawdata/EvtHeader/StereoEvtNumber']
+                    events_per_tel[1] = np.array(eventstream, dtype = dt)
+                    tels_in_file = [2]
+                elif list(file.keys()) == ["MAGIC1_rawdata", "MAGIC2_rawdata"]:
```

Comment:
better use `"MAGIC1_rawdata" in file and "MAGIC2_rawdata" in file`.

Than this has to be the first if